src/dto/event-metadata.dto.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Equals,
+  IsNotEmpty,
+  IsPositive,
+  IsRFC3339,
+  IsUrl,
+  Matches,
+} from 'class-validator';
+/**
+ * @todo add optional attributes
+ * @see https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#overview
+ */
+export class EventMetadataDto {
+  /**
+   * Specification Version
+   * @readonly
+   * @see https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#overview
+   */
+  @Equals(1)
+  readonly specversion;
+
+  /**
+   * Event version
+   * @example 1
+   */
+  @IsPositive()
+  version: number;
+
+  /**
+   * Timestamp of creation date
+   * @example 1524379940
+   */
+  @IsRFC3339()
+  time: number;
```

Comment:
Better to have a date here, more readable

Reply:
it's the standard in https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#time